src/components/dropdown/dropdown.js

```diff
@@ -178,6 +228,9 @@ class BXDropdown extends HTMLElement {
   connectedCallback() {
     this.attachShadow({ mode: 'open' });
     this.render();
+
+    // listen for clicks on the body so we can close the dropdown
+    this.ownerDocument.addEventListener('click', this.#handleFocusIn);
```

Comment:
Seems that my earlier suggestion on bubbling (capturing) `focus` event has to be replaced with `blur`, my apologies for the confusion. Below code on top of yours seems to work instead. Would you try...?

```javascript
#handleFocusOut = event => {
  if (!this.contains(event.relatedTarget)) {
    this.open = false;
  }
};

connectedCallback() {
  ...

  // Detect IE/Edge which have bubbling `window.onfocusout`
  const hasFocusOut = 'onfocusout' in this.ownerDocument.defaultView;
  const focusoutEventName = hasFocusOut ? 'focusout' : 'blur';
  // Use `focusout` if it's there, otherwise use "capture" mode which has similar-to-bubbling effect
  this.ownerDocument.body.addEventListener(focusoutEventName, this.#handleFocusOut, !hasFocusOut);
}
```

Reply:
I'll try that out ... I'm also not sure what happened with that change 😱 definitely not what I intended

Reply:
For `Escape` I still had to handle the `.open = false` in `#handleKeyDownInner` - testing in storybook it doesn't return focus to the dropdown button (I think it puts focus out onto the iframe?), but testing in isolation it behaves as it should.

Reply:
Tab also has some weird behaviour ... `Tab` moves the focus fine, but `shift + Tab` seems to move the focus to the trigger (without activating it's focus state) and then a second `shift + Tab` moves it out of the component